CMakeLists.txt

```diff
@@ -112,22 +130,34 @@ endif()
 ############
 
 find_library(SODIUM_LIBRARY sodium REQUIRED)
-message(STATUS "Using libsodium library at ${SODIUM_LIBRARY}")
+if(SODIUM_LIBRARY)
+  message(STATUS "Using libsodium library at ${SODIUM_LIBRARY}")
+  find_path(SODIUM_INCLUDE_PATH sodium/crypto_verify_32.h)
+  if (SODIUM_INCLUDE_PATH)
+    message(STATUS "SODIUM_INCLUDE_PATH: ${SODIUM_INCLUDE_PATH}")
+  else()
+    message(FATAL_ERROR "Could not find required sodium/crypto_verify_32.h")
+  endif()
+endif()
+
 
 ############
 # HIDAPI
 ############
 
-if(APPLE)
-  include_directories(SYSTEM /usr/include/malloc)
-  if(POLICY CMP0042)
-    cmake_policy(SET CMP0042 NEW)
+option(USE_DEVICE_TREZOR "Trezor hardware wallet suport" ON)
```

Comment:
Again wondering if we can skip this check and assume that support is `ON`.

Would be better to be able to support these dependencies anyway.